crates/fuel-core/src/executor.rs

```diff
@@ -737,8 +761,12 @@ impl Executor {
                 Input::Contract { .. } => {}
                 Input::MessageSigned { message_id, .. }
                 | Input::MessagePredicate { message_id, .. } => {
-                    if let Some(message) = db.storage::<Messages>().get(message_id)? {
-                        if message.fuel_block_spend.is_some() {
+                    if let Some(message) = self
+                        .relayer
+                        .get_message(message_id, &block_da_height)
```

Comment:
I guess it is handled by `message.da_height > block_da_height` and you don't need to pass `block_da_height`

Reply:
Yes but this is more robust because it doesn't rely on the caller to do the check (even though they do)